lib/krb5/fcache.c

```diff
@@ -1162,22 +1169,225 @@ fcc_get_version(krb5_context context,
     return FCACHE(id)->version;
 }
 
+static const char *
+my_basename(const char *fn)
+{
+    const char *base;
+
+    if (strncmp(fn, "FILE:", sizeof("FILE:") - 1))
+        return "";
+    fn += sizeof("FILE:") - 1;
+    for (base = fn + strlen(fn); base >= fn; base--) {
```

Comment:
I would have scanned forward... But this looks correct...

Reply:
Looking for the last component separator is easier to do from the end.

Reply:
(slightly less efficient, yes)

Reply:
I'll change it.